genie-core/src/main/java/com/netflix/genie/core/jobs/workflow/impl/JobTask.java

```diff
@@ -44,20 +44,27 @@
 
     private final AttachmentService attachmentService;
     private final Timer timer;
+    private final GenieFileTransferService fts;
 
     /**
      * Constructor.
      *
      * @param attachmentService An implementation of the Attachment Service
      * @param registry          The metrics registry to use
+     * @param fts               File transfer service
      * @throws GenieException If there is any problem.
      */
     public JobTask(
-        @NotNull final AttachmentService attachmentService,
-        @NotNull final Registry registry
+            @NotNull
+            final AttachmentService attachmentService,
+            @NotNull
+            final Registry registry,
+            @NotNull
+            final GenieFileTransferService fts
```

Comment:
I wonder if we can somehow combine the attachment service and the cache into one as they're pretty similar. Not necessarily required for this feature but a passing thought